pilot/pkg/networking/plugin/authz/rbac.go

```diff
@@ -113,46 +116,57 @@ func (Plugin) OnOutboundCluster(env model.Environment, node model.Proxy, service
 	servicePort *model.Port, cluster *xdsapi.Cluster) {
 }
 
-func isRbacEnabled(store model.IstioConfigStore) bool {
+// isServiceInList checks if a given service of namespace is found in the RbacConfig target list.
+func isServiceInList(svc string, namespace string, li *rbacproto.RbacConfig_Target) bool {
+	for _, ns := range li.Namespaces {
+		if namespace == ns {
+			return true
+		}
+	}
+	for _, service := range li.Services {
+		if service == svc {
+			return true
+		}
+	}
+	return false
```

Comment:
Is this right? Don’t you need to match both service and namespace before returning true ?

Reply:
This is intended. The namespace has a larger scope, the semantic is to include all services in that `namespace` field.

Reply:
Yes i understand that. But the second if implies that when namespace does not match, match by service name atleast. That seems odd. 
You typically match foo.ns.svc.cluster.local or *.ns.svc...

The above logic does *.ns.svc and then foo.*.svc

Reply:
I see, here we're actually using full qualified service name.
So a typical target like this:
```
RbacConfig_Target {
Service: ["product.product-ns.svc.cluster.local", "database.database-ns.svc.cluster.local"]
Namespace: ["default"]
}
```
enables all services in "default" plus another 2 services (`product` and `database`)